packages/typespec-python/src/types.ts

```diff
@@ -173,21 +173,40 @@ function visibilityMapping(visibility?: Visibility[]): string[] | undefined {
     return result;
 }
 
+function clearUsage(emitType: Record<string, any>) {
+    if (emitType.type === "model") {
```

Comment:
why do we have to play with usage here, is tcgc doing something incorrect?

Reply:
If model usage is 0, emitter shall not generate this model. Conversely, if emitter doesn't plan to generate a model, then its usage shall be 0 or changed to 0. I think it is the most clear logic.
Another solution is to add new property for model (e.g. enable_generate) and add logic to filter generated model with the property here: https://github.com/Azure/autorest.python/blob/ff216cbda2e67a2f6849afcf477945d23cef2ca6/packages/typespec-python/generator/pygen/codegen/models/code_model.py#L165

Reply:
i prefer to remove it in property logic, and i doubt why the propagation is needed?

Reply:
For `model FileRequiredMeata extends File {...}`, we need to filter `FileRequiredMeata` and its parent class `File` so propagation is needed.
I see. Let me add a new property to remove the unneeded model

Reply:
Updated.

Reply:
oh, i mean remove them in property handling logic in emitter. sth like:
```
function filterMultipartFileModel(property: SdkPropertyType) {
    // add logic to remove the model from codemodel
}
```

Reply:
Those models may be useful in the future. For example, some models defines `filename` as required, Python SDK could update the docstring for users to hint them even if we don't expose these models. If we filter them in emitter, pygen can't get the info anymore.

Reply:
since python will not use the model directly, i believe these potential meta data will be put in some code model meta.

Reply:
After offline discussion with chenjie, we think current implementation for multipart adds too much magic upon docstring/typing, which will be harder to evolve and maintain. We could add a new type `MultiPartFileType` so that all special logic could be focused on this new type.

Reply:
Sounds good, @msyyc can you open an issue into tcgc for this? thanks!

Reply:
It is specific emitter behavior for python instead of TCGC, and I have updated the PR for review.